app/models/manageiq/providers/workflows/automation_manager/configuration_script_source.rb

```diff
@@ -68,31 +69,34 @@ def sync_from_content(to_delete)
         payload       = File.read(filename)
         workflow      = create_workflow_from_payload(workflow_name, payload, fail_on_invalid_workflow: true)
 
-        to_delete.delete(workflow.name)
+        to_delete.delete(workflow.name) if workflow
       end
     end
   end
 
   def create_workflow_from_payload(name, payload, fail_on_invalid_workflow:)
-    floe_workflow =
+    floe_workflow, payload_error =
       begin
         Floe::Workflow.new(payload)
-      rescue Floe::InvalidWorkflowError
+      rescue Floe::InvalidWorkflowError, NotImplementedError => err
```

Comment:
NOTE there was an issue in floe prior to version v0.15.0 where Map and Parallel raised a `NotImplementedError` rather than a `Floe::InvalidWorkflowError`